helpers/integration.go

```diff
@@ -377,3 +393,17 @@ func CleanUpEphemeralEnvironments(client client.Client, logger *IntegrationLogge
 	logger.LogAuditEvent("Ephemeral environment is deleted and its owning SnapshotEnvironmentBinding is in the process of being deleted", env, LogActionDelete)
 	return nil
 }
+
+func RemoveFinalizer(adapterClient client.Client, logger IntegrationLogger, ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun, finalizer string) error {
+	// Remove the finalizer from this Integration PipelineRun
+	patch := client.MergeFrom(pipelineRun.DeepCopy())
+	controllerutil.RemoveFinalizer(pipelineRun, finalizer)
```

Comment:
The `controllerutil.RemoveFinalizer()` returns a boolean denoting if the finalizer was found and removed, do we want to use that to decide if we need to attempt to patch the resource?

Reply:
Thanks! updated to use the returned boolean from this method before calling Patch.